test/ring_jwt_middleware/core_test.clj

```diff
@@ -203,35 +203,34 @@
     (is (= [] @log-events))
     (reset-log-events))
 
-  (with-redefs
-    [time/now (constantly (time/date-time 2017 02 16 14 14 11))]
-    (is (= '("This JWT has expired since 1s (we don't allow JWT older than 1 day; we only checked creation date and not maximal expiration date)")
-           (sut/validate-jwt "jwt" decoded-jwt-2 86400 test-log-fn))))
-
-  (with-redefs
-    [time/now (constantly (time/date-time 2017 02 16 15 14 10 0))]
-    (is (= '("This JWT has expired since 1h (we don't allow JWT older than 1 day; we only checked creation date and not maximal expiration date)")
-           (sut/validate-jwt "jwt" decoded-jwt-2 86400 test-log-fn))))
-
-  (with-redefs
-    [time/now (constantly (time/date-time 2017 02 17 15 14 10 0))]
-    (is (= '("This JWT has expired since 1 day 1h (we don't allow JWT older than 1 day; we only checked creation date and not maximal expiration date)")
-           (sut/validate-jwt "jwt" decoded-jwt-2 86400 test-log-fn))))
-
-  (with-redefs
-    [time/now (constantly (time/date-time 2017 02 18 15 14 10 0))]
-    (is (= '("This JWT has expired since 2 days 1h (we don't allow JWT older than 1 day; we only checked creation date and not maximal expiration date)")
-           (sut/validate-jwt "jwt" decoded-jwt-2 86400 test-log-fn))))
-
-  (with-redefs
-    [time/now (constantly (time/date-time 2019 04 03 8 24 5 123))]
-    (is (= '("This JWT has expired since 2 years 45 days 18h 9min 55s (we don't allow JWT older than 1 day; we only checked creation date and not maximal expiration date)")
-           (sut/validate-jwt "jwt" decoded-jwt-2 86400 test-log-fn))))
-
-  (with-redefs
-    [time/now (constantly (time/date-time 2017 02 16 14 14 11))]
-    (is (= '("This JWT has expired since 1s (we don't allow JWT older than 1 day; we only checked creation date and not maximal expiration date)")
-           (sut/validate-jwt "jwt" decoded-jwt-2 86400 nil test-log-fn)))))
+  (testing "expiration message"
+    (testing "expired time"
+      (let [explain-msg "This JWT has expired %s ago (we don't allow JWT older than 1 day; we only checked creation date and not maximal expiration date)"
+            tst-fn (fn [d expected]
```

Comment:
I recommend applying @gbuisson's change suggestion in the messages in the tests too.